### PR TITLE
Create infraattributesprocessor go module

### PR DESCRIPTION
**Pull request by jackgopack4**
### What does this PR do?

creates a go module for comp/otelcol/otlp/components/processor/infraattributesprocessor

### Motivation

Closes [OTEL-1995](https://datadoghq.atlassian.net/browse/OTEL-1995)

### Additional Notes

It appears that a dependency was added to serverless.

### Possible Drawbacks / Trade-offs

one more go module to maintain in agent codebase (discussions to be had about how we add and decide how to support these go modules in the future)

### Describe how to test/QA your changes

no code changes so unit/integrations should be sufficient

[OTEL-1995]: https://datadoghq.atlassian.net/browse/OTEL-1995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: comp/otelcol/otlp/collector_test.go [hunk 1]
```diff
@@ -103,7 +104,7 @@ func TestStartPipelineFromConfig(t *testing.T) {
 		{path: "receiver/advanced.yaml"},
 		{
 			path: "receiver/typo.yaml",
-			err:  "error decoding 'receivers': error reading configuration for \"otlp\": 1 error(s) decoding:\n\n* 'protocols' has invalid keys: htttp",
+			err:  "'protocols' has invalid keys: htttp",
```
Comment (songy23) [inline comp/otelcol/otlp/collector_test.go:107]: Why this change?

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go [hunk 2]
```diff
@@ -19,13 +19,15 @@ import (
 var processorCapabilities = consumer.Capabilities{MutatesData: true}
 
 type factory struct {
-	tagger taggerClient
+	tagger     taggerClient
+	generateID GenerateKubeMetadataEntityID
```
Comment (songy23) [inline comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go:23]: I assume this is needed because `comp/core/workloadmeta/collectors/util` is not a module?

Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go:23]: correct, workaround for now to avoid creating half a dozen other modules

**Review by purple4reina — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by chouetz — APPROVED** [review]
OK for devx-infra files

File: comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go (hunk 2 above)
Comment (jackgopack4) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go:23]: understood; Dinesh suggested this approach but if you want me to get back to the other dependencies first happy to do that

Comment (songy23) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go:23]: I'm OK with the current approach, it's an incremental step. Can you leave a TODO here to remove generateID as a dependency in the future?

Comment (dineshg13) [reply comp/otelcol/otlp/components/processor/infraattributesprocessor/factory.go:23]: I would recommend to take this incremental step as priority of OCB is unclear at the moment.

File: cmd/serverless/dependencies_linux_amd64.txt [hunk 3]
```diff
@@ -438,6 +438,7 @@ github.com/go-logr/logr
 github.com/go-logr/logr/funcr
 github.com/go-logr/stdr
 github.com/go-viper/mapstructure/v2
+github.com/go-viper/mapstructure/v2/internal/errors
```
Comment (dustmop) [inline cmd/serverless/dependencies_linux_amd64.txt:441]: Where are these error types being used? I'm pretty sure thing change is okay, but I ask because we're working on refactoring the config to not use mapstructure in viper anymore, so it would be good to know what's changing here.

Comment (jackgopack4) [reply cmd/serverless/dependencies_linux_amd64.txt:441]: good question, I'll look into this. I believe that `go mod tidy` pulled in v2.1 of go-viper/mapstructure over v2.0.0 alpha, which also caused this indirect dependency to come down.

Comment (dustmop) [reply cmd/serverless/dependencies_linux_amd64.txt:441]: Ok, if it's just internal and not used by the code then that's fine. Just wanted to know if any code was using it.

**Review by dustmop — APPROVED** [review]
LGTM

File: cmd/serverless/dependencies_linux_amd64.txt (hunk 3 above)
Comment (jackgopack4) [reply cmd/serverless/dependencies_linux_amd64.txt:441]: - [go.opentelemetry.io/collector/confmap](go.opentelemetry.io/collector/confmap) imports [github.com/go-viper/mapstructure/v2@v2.1.0](github.com/go-viper/mapstructure/v2@v2.1.0)
- [collector/confmap](go.opentelemetry.io/collector/confmap) also imports [github.com/knadh/koanf](github.com/knadh/koanf) which also imports [mapstructure/v2@v2.1.0](github.com/go-viper/mapstructure/v2@v2.1.0)

None of our packages or direct dependencies import [github.com/go-viper/mapstructure/v2/internal/errors](github.com/go-viper/mapstructure/v2/internal/errors); rather [mapstructure/v2/internal/errors](github.com/go-viper/mapstructure/v2/internal/errors) is a sub-module of [go-viper/mapstructure](github.com/go-viper/mapstructure/v2@v2.1.0) so I don't know if there's a way to *not* import it

**Review by dineshg13 — APPROVED** [review]

**Review by songy23 — APPROVED** [review]
